code/modules/clothing/glasses/glasses.dm

```diff
@@ -543,3 +543,95 @@ var/list/science_goggles_wearers = list()
 	darkness_view = -1
 	prescription = 1
 	eyeprot = 1
+
+//////////////////
+
+
+/obj/item/clothing/glasses/emitter
+	name = "emitter goggles"
+	desc = "Become literally unable to stop firing beams."
+	icon_state = "emitter"
+	item_state = "glasses"
+	origin_tech = Tc_POWERSTORAGE + "=5;" + Tc_MATERIALS + "=3" + Tc_ANOMALY + "=4"
+	species_fit = list(GREY_SHAPED, INSECT_SHAPED)
+	var/mob/living/emitter
+	var/obj/effect/beam/emitter/eyes/beam
+	var/previous_dir
+	var/turf/previous_loc
+
+/obj/item/clothing/glasses/emitter/equipped(var/mob/M, var/slot)
+	..()
+	if(slot == slot_glasses)
+		emitter = M
+		enable(emitter)
+		processing_objects.Add(src)
+
+/obj/item/clothing/glasses/emitter/unequipped(var/mob/M, var/from_slot)
+	..()
+	previous_dir = null
+	previous_loc = null
+	if(from_slot == slot_glasses)
+		disable(emitter)
+		processing_objects.Remove(src)
+
+
+/obj/item/clothing/glasses/emitter/Destroy()
+	disable(emitter)
+	processing_objects.Remove(src)
+	previous_dir = null
+	previous_loc = null
+	..()
+
+/obj/item/clothing/glasses/emitter/proc/enable(var/mob/living/mob)
+	if (istype(emitter))
+		emitter.callOnStartMove |= "\ref[src]"
+		emitter.callOnStartMove["\ref[src]"] = "update_emitter_start"
+		emitter.callOnEndMove |= "\ref[src]"
+		emitter.callOnEndMove["\ref[src]"] = "update_emitter_end"
```

Comment:
`|=` is slow. I think you can condense this with just `L["\ref[src]"] = "update_emitter_start"`, since `"\ref[src]"` and `"update_emitter_start"` are just strings (doing this with atoms as indexes causes issue for some reasons, I learned it in the runechat PR).